source/OptimaJet.DWKit.StarterApplication/Services/HttpCurrentUserContext.cs

```diff
@@ -26,5 +68,69 @@ public class HttpCurrentUserContext : ICurrentUserContext
             }
         }
 
+        public string Email {
```

Comment:
If the current user is pulled in to this class, then each of these properties can be proxies to the memoized current user which would greatly reduce the amount of memoization in this class (which would achieve simpler implementation, I think)

Reply:
I explained earlier that it would be an inappropriate coupling to bring in model classes into this class.  This is low level access to data in HttpContent (or derived from it via Auth0 Management API).  It was a conscience decision to layer it this way.

Reply:
ok, I just mean that as-is, it's kinda messy. And because this is the `CurrentUserContext`, I thought the data should be primarily backed by a current user, rather than the `HttpContext` (which contains all info about a request). 

The memoization seems to just be a bit much here. 
In practice, these properties should only be read once, because (and correct me if I'm wrong, I could be totally off on intent here):

request comes in:
 - user has a user record
     - CurrentUser should be looked up via the db, and this class's functionality (as it is currently) is then not used, because all the data exists in the db, and is maintained there

 - user does not yet have a user record
     - This class is used to build the User Entity before saving, in which case all of these attributes, Email, Given Name, etc, are only read once, and memoization is unneeded.